### ISSUE TITLE
Pages not loading - Admin - Manage Streams - Settings

### ISSUE THREAD

**Issue by ezeri** [zulip/zulip-desktop#52]
The following pages :
- Administration
- Manage Streams
- Settings

Alll show the loading icon for 10-15 seconds and fail to load, without any message in the terminal.

Using CentOs 6.7

Looked into the latest releases of zulip-desktop, and no modifications have been done that would affect this.

We do not have the problem under Mac though. Only the Linux version.

Cheers,

Belisaire.

![zulip](https://cloud.githubusercontent.com/assets/12232340/19436513/cf5dcd7c-9470-11e6-96ff-d00687036c83.jpg)

**Comment by timabbott** [comment]
Can you try using https://github.com/zulip/zulip-electron?  That's intended to replace this project soon.

**Comment by timabbott** [comment]
(Sorry for the slow reply!)

**Comment by ezeri** [comment]
Sorry, was on vacation myself when you answered, so only getting around to this now. I will start tests with electron.

Just before I went on vacation, we encountered another weird problem. With certain artists on certain machines, a signal is passed on to zulip that provokes a crash in Qt5 (Gotten the info through system logs), that have crashed the X server of the machine. Crashing the sesion completely. Instantaneously and without any warning.

We deduce that it is Zulip-desktop as we did not have Qt5 before installing zulip and it is Qt5 that is crashing.

I have not looked into it much more, so far.

**Comment by ezeri** [comment]
Sorry, closed by accident. -_-'

**Comment by ezeri** [comment]
I have worked around the lib problems. 

I have a version up and running, even if I had to deactivage the TLS verification for nodejs. The pages "Administration", "Manage Streams" and "Settings" are working on zulip-electron.

I get errors with the notification system though.

```
(electron:2745241): libnotify-CRITICAL **: notify_notification_new: assertion `attach == NULL || GTK_IS_WIDGET (attach)' failed

(electron:2745241): libnotify-CRITICAL **: notify_notification_add_action: assertion `notification != NULL' failed

(electron:2745241): libnotify-CRITICAL **: notify_notification_set_image_from_pixbuf: assertion `notification != NULL' failed

(electron:2745241): libnotify-CRITICAL **: notify_notification_set_timeout: assertion `notification != NULL' failed

(electron:2745241): libnotify-CRITICAL **: notify_notification_show: assertion `notification != NULL' failed

(electron:2745241): libnotify-CRITICAL **: notify_notification_close: assertion `notification != NULL' failed
```

Also, Unlike the zulip-desktop, the app is not running in background. If I close the window, the zulip app is closed completely.

**Comment by akashnimare** [comment]
> Also, Unlike the zulip-desktop, the app is not running in the background. If I close the window, the zulip app is closed completely.

I think this is the expected behavior on Windows and Linux systems whereas on OSX it will keep running in the background.

**Comment by timabbott** [comment]
@akashnimare do you think we should change the Linux/Windows behavior (or add an option to let us do this differently)?  I feel like people may want to be able to close the window but then still receive desktop notifications.

Feedback from users on this question would be helpful!

**Comment by akashnimare** [comment]
@timabbott I think we should not change the default behavior which is closing the app on clicking ❎  button. I could add an extra option somewhere in our menu items (maybe just above **Quit Zulip** menu) which will hide the app window so that users will be able to receive desktop notifications.

**Comment by timabbott** [comment]
yeah, that's probably reasonable.

**Comment by ezeri** [comment]
Hello guys,

From my point of view. When you look at Skype, Pidgin, the default behavior is to close the current window of the application and to keep it running in the background, to be able to unclutter your desktop without losing the notification abilities. For me the current behavior of zulip-electron is non-standard for an application that allows you to have this type of notification.

Personally I would prefer seeing an option that changes the behavior of the :x:, like that the user can change how the app wil behave, but he won't have to go to "Minimize app" each time he wants to remove the window that is open.

Cheers,

Belisaire.

**Comment by timabbott** [comment]
Yeah, I agree a setting that changes the behavior would be a lot better than the menu item proposal.

**Comment by akashnimare** [comment]
@ezeri Thanks for the feedback. Moving this discussion to https://github.com/zulip/zulip-electron/issues/84
